### DISCUSSION TITLE
Can I run any deepspeed application with torch.distributed.run?

### DISCUSSION THREAD

**Discussion by mathewjacob1002** [deepspeedai/DeepSpeed#3887] [Q&A]
I was interested in learning that, with deepspeed, you can use torch.distributed.run to run HuggingFace and Pytorch Lightning application with deepspeed optimizations. Is this also true for any arbitrary deepspeed application? Additionally, I was also curious as to what the differences were between the deepspeed launcher and using the torch.distribued.run launcher?

Thank you so much!

**Comment by mzamini92** [comment]
The deepspeed launcher is a wrapper around torch.distributed.run that provides additional features. deepspeed launcher uses the deepspeed runtime, which some provides a number of optimizations like: 
Pipeline parallelism: This optimization allows multiple models to be executed in parallel on the same GPU.
Model parallelism: This optimization allows a single model to be executed in parallel on multiple GPUs.
Tensor fusion: This optimization merges multiple small tensors into larger tensors, which can improve memory bandwidth and cache efficiency.
